src/enigma2/Epg.cpp

```diff
@@ -521,3 +527,41 @@ bool Epg::LoadInitialEPGForGroup(const std::shared_ptr<ChannelGroup> group)
   Logger::Log(LEVEL_INFO, "%s Loaded %u EPG Entries for group '%s'", __FUNCTION__, iNumEPG, group->GetGroupName().c_str());
   return true;
 }
+
+void Epg::UpdateTimerEPGFallbackEntries(const std::vector<enigma2::data::EpgEntry> &timerBasedEntries)
+{
+  CLockObject lock(m_mutex);
+  time_t now = time(nullptr);
+  time_t until = now + m_epgMaxDaysSeconds;
+
+  m_timerBasedEntries.clear();
+
+  for (auto& timerBasedEntry : timerBasedEntries)
+  {
+    if (timerBasedEntry.GetEndTime() < now || timerBasedEntry.GetEndTime() > until)
+      m_timerBasedEntries.emplace_back(timerBasedEntry);
+  }
+}
+
+int Epg::TransferTimerBasedEntries(ADDON_HANDLE handle, int epgChannelId)
+{
+  int numTransferred = 0;
+
+  CLockObject lock(m_mutex);
+  for (auto& timerBasedEntry : m_timerBasedEntries)
+  {
+    if (epgChannelId == timerBasedEntry.GetChannelId())
+    {
+      EPG_TAG broadcast;
+      memset(&broadcast, 0, sizeof(EPG_TAG));
```

Comment:
`EPG_TAG broadcast = {0};` is shorter and in differnece to `memset` code will also work if `EPG_TAG` should get a vtable for whatever reason. `memset` would zero the vtable, which is... well, not a good idea.

Reply:
Cool, memset if used a lot of the existing codebase. I'll do a once through and fix them all.

Reply:
Done